### PR TITLE
feat: Add last_write_time and table buffer size

### PR DESCRIPTION
**Pull request by pauldix**
This adds tracking of the instant of the last write to open buffer segment and methods to the table buffer to compute the estimated memory size of it.

These will be used by a background task that will continuously check to see if tables should be persisted ahead of time to free up buffer memory space.

Originally, I had hoped to have the size tracking happen as the buffer was built so that returning the size would be zero cost (i.e. just returning a value), but I found in different kinds of testing that I wasn't able to get something that was even close to accurate. So for now it will use this more expensive computed method and we'll check on this periodically (every couple of seconds) to see when to persist.

This is the first bit of work for #24956

### CODE REVIEW

File: influxdb3_write/src/write_buffer/table_buffer.rs [hunk 1]
```diff
@@ -231,6 +232,17 @@ impl TableBuffer {
 
         Ok(RecordBatch::try_new(schema, cols)?)
     }
+
+    /// Returns an estimate of the size of this table buffer based on the data and index sizes.
+    #[allow(dead_code)]
```
Comment (mgattozzi) [inline influxdb3_write/src/write_buffer/table_buffer.rs:237]: Why is this here? Is this like that because it's going to be used in a later PR?

**Review by mgattozzi — COMMENTED** [review]
Just a question below before I approve 😁

File: influxdb3_write/src/write_buffer/table_buffer.rs (hunk 1 above)
Comment (pauldix) [reply influxdb3_write/src/write_buffer/table_buffer.rs:237]: Yeah, the lint was complaining

Comment (pauldix) [reply influxdb3_write/src/write_buffer/table_buffer.rs]: Updated with name change to have `_` prefix to avoid the lint and removed the allow.

**Review by mgattozzi — APPROVED** [review]

**Pull request merged**